Shrink GCE bootstrap image to minimum size, and auto-expand it to actual size on first boot.

clefru opened the pull request:
###### Motivation for this change
The standard nixops GCE image is outdated (16.03) and too big (100G). The latter means that nixops deployments can't make use of the free tier that GCP introduced this month, which gives 30GB disk space for free. See https://github.com/NixOS/nixops/issues/590
 
###### Things done
I used nixos/maintainers/scripts/gce/create-gce.sh to build an image. I brought it up on GCE to see whether it boots and root disk resizing works, and from my testing it does. I had to cherry-pick this into my build repository https://github.com/NixOS/nixpkgs/pull/24143 otherwise QEMU fails when building the image.

- [ ] Tested using sandboxing
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

clefru committed:
Make GCE image as small as possible and incorporate partition growing
when users of nixops specified a larger root disk via
deployment.gce.rootDiskSize

1GB is the smallest possible size as GCP doesn't support
fractions of GB for RAW images, see
https://cloud.google.com/compute/docs/images/import-existing-image#requirements

ljli reviewed (commented):
Nice!
#24267 also addresses kernel.yama.ptrace_scope, just for reference.
Also I had to resize the filesystem manually, we might want to configure `fileSystems."/".autoResize = true` for the bootstrap image.

fpletz commented on nixos/modules/virtualisation/google-compute-image.nix:
```diff
@@ -261,7 +262,7 @@ in
 
     # set ptrace protections
-    "kernel.yama.ptrace_scope" = mkDefault "1";
+    "kernel.yama.ptrace_scope" = mkOverride 999 "1";
```
Using 500 like in #24267 makes more sense IMHO.

clefru replied:
You are right. I dropped the commit change this line.

clefru commented:
@ljli: Thanks, I missed in my testing that autoResize=true is required. Added to the PR.

(Also I moved to filesystem.'/'.device="/dev/disk/by-label" to match the style of other *-image.nix files)

Mic92 commented:
Thanks!
